app/Http/Controllers/Api/MovieController.php

```diff
@@ -26,40 +26,31 @@ public function index()
      */
     public function store(Request $request)
     {
-        //
+        return Movie::create($request->all());
     }
 
-    /**
-     * Display the specified resource.
-     *
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
     public function show($id)
     {
-        return Movie::findOrFail($id);
+        return Movie::with('genre')->findOrFail($id);
     }
 
-    /**
-     * Update the specified resource in storage.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
     public function update(Request $request, $id)
     {
-        //
+        $movie = Movie::findOrFail($id);
+        $movie->update($request->all());
+        return $movie;
     }
 
-    /**
-     * Remove the specified resource from storage.
-     *
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
     public function destroy($id)
     {
         //
     }
+
+    public function viewed($id)
```

Comment:
cemu sluzi ruta koja ovo gadja?
ako je da broji koliko ima pregleda, to bi trebalo nakaciti negde drugde, tipa na middleware za rutu /movie/ID, jer ovako ti 2 puta dobavljas movie samo da bi se incrementovao counter